sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -906,26 +906,25 @@ def delayed_bundle_application(self,
     # type: (...) -> beam_fn_api_pb2.DelayedBundleApplication
     assert op.input_info is not None
     # TODO(SDF): For non-root nodes, need main_input_coder + residual_coder.
-    ((element_and_restriction, output_watermark),
-     deferred_watermark) = deferred_remainder
-    if deferred_watermark:
-      assert isinstance(deferred_watermark, timestamp.Duration)
+    element_and_restriction = deferred_remainder.windowed_value
```

Comment:
You can still use tuple unpacking here, e.g. 

`(element_and_restriction, deferred_timestamp, current_watermark) = deferred_remainder`